source/MaterialXGenOsl/OslSyntax.cpp

```diff
@@ -246,6 +246,68 @@ class OSLMatrix3TypeSyntax : public AggregateTypeSyntax
     }
 };
 
+class FileFormatSyntax : public AggregateTypeSyntax
+{
+  public:
+    FileFormatSyntax(const string& name, const string& defaultValue, const string& uniformDefaultValue,
+                     const string& typeAlias = EMPTY_STRING, const string& typeDefinition = EMPTY_STRING,
+                     const StringVec& members = EMPTY_MEMBERS) :
+        AggregateTypeSyntax(name, defaultValue, uniformDefaultValue, typeAlias, typeDefinition, members)
+    {}
+    
+    string getValue(const ShaderPort* port, bool uniform) const override
+    {
+        if (!port)
+        {
+            return EMPTY_STRING;
+        }
+
+        StringVec values;
+        values.push_back(port->getValue() ? port->getValue()->getValueString() : EMPTY_STRING);
+        values.push_back(port->getColorspace());
+        return getValue(values, uniform);
+    }
+
+    string getValue(const Value& value, bool uniform) const override
+    {
+        StringVec values;
+        values.push_back(value.getValueString());
+        values.push_back(EMPTY_STRING); // TODO: How to get color space initialized
+
+        return getValue(values, uniform);
+    }
+
+    string getValue(const StringVec& values, bool uniform) const override
```

Comment:
This method could be removed if you construct the value strings directly above instead.

Reply:
Looking at this:
I'm not sure where this is called from but it's an override on the parent class AggregateTypeSyntax.
If we don't override it then the wrong result occurs. I'm not sure when this version is every used though.

Reply:
Aha sorry I didn't notice this was virtual. Then you should keep it in :)